generator/lib/src/entity_resolver.dart

```diff
@@ -87,6 +90,18 @@ class EntityResolver extends Builder {
 
         final _idAnnotation = _idChecker.firstAnnotationOfExact(f);
         propUid = _idAnnotation.getField('uid').toIntValue();
+      } else if (_uniqueChecker.hasAnnotationOfExact(f)) {
+        final _uniqueAnnotation = _uniqueChecker.firstAnnotationOfExact(f);
+        propUid = _uniqueAnnotation.getField('uid').toIntValue();
+        indexUid = _uniqueAnnotation.getField('indexUid').toIntValue();
+        fieldType = _uniqueAnnotation.getField('type').toIntValue();
+        flags = OBXPropertyFlag.UNIQUE;
+      } else if (_indexChecker.hasAnnotationOfExact(f)) {
+        final _indexAnnotation = _indexChecker.firstAnnotationOfExact(f);
+        propUid = _indexAnnotation.getField('uid').toIntValue();
+        indexUid = _indexAnnotation.getField('indexUid').toIntValue();
+        fieldType = _indexAnnotation.getField('type').toIntValue();
+        flags = OBXPropertyFlag.INDEXED;
```

Comment:
Hm, I see why it would make sense to have `@Unique` and `@Index` be exclusive annotations. However, they may be used both on one property. Also, as said on `test/entity.dart` they should not replace `@Property`.

Compare with Java: https://docs.objectbox.io/entity-annotations#property-indexes-with-index ff
The index type may be further configured, even if using `@Unique`. But we/I can also look into implementing this.

Reply:
I'm out of my depth here...

Reply:
outdated conversation. done like in Java (separate annotations, index type), resolving the conversation.